src/app/api/organizations/[organizationId]/route.ts

```diff
@@ -24,3 +27,43 @@ export async function DELETE(
   }
   return new NextResponse(undefined, { status: 204 });
 }
+
+// @route PUT /api/organizations/[organizationId] - updates an existing organization in the database
+export async function PUT(
+  _request: NextRequest,
+  { params }: { params: { organizationId: string } }
+) {
+  try {
+    const validationResult = zObjectId.safeParse(params.organizationId);
+    if (!validationResult.success) {
+      throw new Error('Invalid Organization Id');
+    }
+
+    // Get the updated organization data
+    const requestBody = await _request.json();
```

Comment:
we'll want to validate the request body to make sure it is in the right format before sending it to the server func. Here's an example, except this will be validating against zUpdateOrganizationRequest. One thing to add to that (and I just learned this recently!) is that the return value of the safeParse function has a success prop and a data prop. The data prop is what we should be passing into the server func, as it strips away any extra fields (extra fields do not throw validation errors, they are just not validated). Jun has a good example of that [here](https://github.com/hack4impact-utk/compassion-ministries/pull/32/files), `src/app/api/organizations/route.ts` lines 11-18